signer/core/api.go

```diff
@@ -449,23 +476,49 @@ func logDiff(original *SignTxRequest, new *SignTxResponse) bool {
 	return modified
 }
 
+func (api *SignerAPI) lookupPassword(address common.Address) (string, error) {
+	return api.credentials.Get(address.Hex())
+}
+
+func (api *SignerAPI) lookupOrQueryPassword(address common.Address, title, prompt string) (string, error) {
+	// Look up the password and return if available
+	if pw, err := api.lookupPassword(address); err == nil {
+		return pw, nil
+	}
+	// Password unavailable, request it from the user
+	pwResp, err := api.UI.OnInputRequired(UserInputRequest{title, prompt, true})
+	if err != nil {
+		log.Warn("error obtaining password", "error", err)
+		// We'll not forward the error here, in case the error contains info about the response from the UI,
+		// which could leak the password if it was malformed json or something
+		return "", errors.New("internal error")
+	}
+	return pwResp.Text, nil
+}
+
 // SignTransaction signs the given Transaction and returns it both as json and rlp-encoded form
 func (api *SignerAPI) SignTransaction(ctx context.Context, args SendTxArgs, methodSelector *string) (*ethapi.SignTransactionResult, error) {
 	var (
 		err    error
 		result SignTxResponse
+		msgs   *ValidationMessages
 	)
-	msgs, err := api.validator.ValidateTransaction(&args, methodSelector)
-	if err != nil {
-		return nil, err
+	if args.IsPrivate {
+		msgs = new(ValidationMessages)
+		msgs.Info(fmt.Sprintf("Quorum private transaction. Key: %s", args.Data))
```

Comment:
Is it necessary to leak private payload to validation message?

Reply:
Done